### PR TITLE
Fix file uploads saving in case generated by `MultipartFormDataParser`

### PR DESCRIPTION
**Pull request by hyuzea**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #17037

### CODE REVIEW

**Comment by sup-ham** [comment]
Also the docs need to be adjusted.

**Comment by samdark** [comment]
@sup-ham which docs?

**Comment by samdark** [comment]
There seems to be trouble with the changes under Windows.

**Comment by My6UoT9** [comment]
Would this work if my filename is "@webroot.jpg" ? Looks like trouble?

File: framework/web/UploadedFile.php [hunk 1]
```diff
@@ -149,25 +162,48 @@ public static function reset()
...
+        return !$deleteTempFile || $this->deleteTempFile();
+    }
+
+    /**
+     * @return bool|int
+     */
+    private function copyTempFile($file)
+    {
+        if (!is_resource($this->_tempResource)) {
+            return is_file($this->tempName) && copy($this->tempName, $file);
```
Comment (samdark) [inline framework/web/UploadedFile.php]: That looks suspicious to me. Previously there was `is_uploaded_file` check that ensured that file moved is what came via POST. Now, resource is empty only when the method is POST but the check is missing.

**Review by samdark — CHANGES_REQUESTED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

**Pull request merged**